### ISSUE TITLE
Outdated tutorials

### ISSUE THREAD

**Issue by kadyb** [r-spatial/evolution#20]
When the packages will be retired, many popular online tutorials will become obsolete. For example, if we write: "*how to load vector data (or shapefile) in R*" in Google, the top 3 results (at least for me) are:

```
1. https://www.neonscience.org/resources/learning-hub/tutorials/intro-vector-data-r
2. http://geoprofesja.pl/en/loading-and-saving-spatial-vector-layers-in-r/
3. https://www.earthdatascience.org/courses/earth-analytics/spatial-data-r/intro-vector-data-r/
```

and all use `{rgdal}`. The authors probably won't be interested in updating tutorials to `{sf}` or `{terra}`, but I think it would be worth taking some action, e.g. series of blogposts on R-Spatial and their wide promotion?

**Comment by rsbivand** [comment]
It would be worth contacting the authors or their sponsors to point out their failure to keep up. Listing their pageranks to track the desired decline migjt aldo be worthwhile. Please also consider other languages, so we'd need help with search terms in Chinese, Spanish, French, Vietnamese, Indonesian, and others. Who might we ping?

**Comment by rsbivand** [comment]
Maybe we need a short blog or film showing how to transition, to offer to authors to add at the start of their blogs?

**Comment by kadyb** [comment]
It would be ideal to update the most popular tutorials, but personally I think this is impossible. Most of them were written a few years ago and I suspect that most authors no longer work in this domein. If the tutorials / courses were created with funding, I think there will be no interest in updating them either.

Here is a good (and widely appreciated) tutorial from @Nowosad: https://geocompx.org/post/2023/rgdal-retirement/ But there is still the question of whether any author will be interested in updating or whether we should focus on creating and promoting new tutorials.

**Comment by kadyb** [comment]
The open issue on updating materials on [The R Graph Gallery](https://r-graph-gallery.com/map.html): https://github.com/holtzy/R-graph-gallery/issues/95

**Comment by kadyb** [comment]
@edzer, would you be interested in publishing a blogpost on R-Spatial about "How to load and save vector data in R"? I can write a short tutorial on how to load e.g. shp, zipped shp, gpkg, geojson in `{sf}`?

**Comment by florisvdh** [comment]
> @edzer, would you be interested in publishing a blogpost on R-Spatial about "How to load and save vector data in R"? I can write a short tutorial on how to load e.g. shp, zipped shp, gpkg, geojson in `{sf}`?

:+1: Feel free to borrow from the tutorials website of my organization, where two formats have been specifically recommended (in 2020):

- [How to use open vector file formats in R: GeoPackage & GeoJSON](https://inbo.github.io/tutorials/tutorials/spatial_standards_vector/)
- [Meet some popular open geospatial standards!](https://inbo.github.io/tutorials/articles/geospatial_standards/)

BTW I'm aware that some of our geospatial tutorials are in need of an update (I intend to update them).

Also I believe that GeoParquet will deserve its place in this respect. Probably also DuckDB, which is [rumored on the CNG blog](https://cloudnativegeo.org/blog/2023/09/duckdb-the-indispensable-geospatial-tool-you-didnt-know-you-were-missing/) to to be a potential successor of GeoPackage, or maybe replace its SQLite backend.

**Comment by edzer** [comment]
Yes, very welcome. @florisvdh I believe DuckDB is more like a software layer than a file format; the blog talks consistently about analysing (geo)parquet files. In that sense, there's a certain overlap with arrow. A blog in that direction would also be very welcome.

**Comment by florisvdh** [comment]
> I believe DuckDB is more like a software layer than a file format; the blog talks consistently about analysing (geo)parquet files.

Indeed the blog focuses on this ability:

>> The coolest thing about it is that you can just treat any Parquet file or set of Parquet files as a table, without actually ‘importing’ it into the database.

>> You can also easily write out new Parquet files, with lots of great options like controlling the size of row groups and doing partitioned writes. And you can do this all in one call, without ever instantiating a table.

But I think it can also just store tables in its file format.

>> Since the entire database is just a location on disk I can easily start up the database and inspect it to confirm what I did programmatically. It’s also super easy to move your database, you just move it like any other file, and connect in the new location. The other thing I really appreciate is you can easily see how much space your database is taking up – since it’s just a file you can view it and delete it just like others.

**Comment by florisvdh** [comment]
Adding a few leads wrt DuckDB in R; there may be more.

There's {[duckdbfs](https://cboettig.github.io/duckdbfs)} which centers on accessing 'file paths, URLs, or S3 URIs'. Has a spatial section on homepage: https://cboettig.github.io/duckdbfs/#spatial-data; there's also comparison with {arrow}.

There's also {[duckplyr](https://duckdblabs.github.io/duckplyr/)} which uses the DuckDB processing abilities when using {dplyr} verbs.

**Comment by kadyb** [comment]
I have the first version of the tutorial about loading and saving data in `sf`? What do you think?

https://github.com/kadyb/sf_load_save

**Comment by kadyb** [comment]
@edzer, would you accept publishing my tutorial on the R-spatial blog?

**Comment by florisvdh** [comment]
> I have the first version of the tutorial about loading and saving data in `sf`? What do you think?
> 
> https://github.com/kadyb/sf_load_save

I've read your tutorial @kadyb; very didactical, reads well! Dropped some suggestions in https://github.com/kadyb/sf_load_save/issues/2.

**Comment by edzer** [comment]
> would you accept publishing my tutorial on the R-spatial blog?

Sure, that's an option, an alternative might be a tutorial in `sf` as this only concerns `sf`, right? Which one would be more widely used?

**Comment by kadyb** [comment]
> Sure, that's an option, an alternative might be a tutorial in `sf` as this only concerns `sf`, right? Which one would be more widely used?

I don't have strong preference, but there is already one tutorial about loading and saving data (https://r-spatial.github.io/sf/articles/sf2.html). I don't see the point in duplicating it (and too many vignettes can also be overwhelming for new users). Nevertheless, the vignette will probably be more popular.
